python/tests/stress/arcticdb/version_store/test_mem_leaks.py

```diff
@@ -146,13 +160,174 @@ def grow_exp(df_to_grow: pd.DataFrame, num_times_xx2: int):
+    df = get_sample_dataframe(size)
+    df.index = pd.date_range(start="2000-1-1", periods=size, freq="s")
+    df.index.name = "timestamp"
+    return df
+
+
+def query_filter_then_groupby_with_aggregations() -> QueryBuilder:
+    """
+    groupby composite aggregation query for QueryBuilder memory tests.
+    The query basically will do aggregation of half of dataframe
+    """
+    q = QueryBuilder()
+    return (
+        q[q["bool"]]
+        .groupby("uint8")
+        .agg({"uint32": "mean", 
+              "int32": "sum", 
+              "strings": "count", 
+              "float64": "sum", 
+              "float32": "min", 
+              "int16": "max"})
+    )
+
+def query_no_filter_only_groupby_with_aggregations() -> QueryBuilder:
+    """
+    groupby composite aggregation query for QueryBuilder memory tests.
+    The query basically will do aggregation of half of dataframe
+    """
+    q = QueryBuilder()
+    return (
+        q.groupby("uint8")
+        .agg({"uint32": "mean", 
+              "int32": "sum", 
+              "strings": "count", 
+              "float64": "sum", 
+              "float32": "min", 
+              "int16": "max"})
+    )
+
+
+def query_filter_impossible_cond_groupby_with_aggregations_for_whole_frame() -> QueryBuilder:
+    """
+    groupby composite aggregation query for QueryBuilder memory tests.
+    The query basically will do aggregation of whole dataframe
+    """
+    q = QueryBuilder()
+    return (
+        q[q["strings"] != "QASDFGH"]
+        .groupby("int16")
+        .agg({"uint32": "mean", 
+              "int32": "sum", 
+              "strings": "count", 
+              "float64": "sum", 
+              "float32": "min", 
+              "int16": "max"})
+    )
+
+
+def query_apply_clause_only(strng: str) -> QueryBuilder:
+    """
+    Apply query for QueryBuilder memory tests.
+    This version of apply does couple of nested operations
+    with columns that were just added
+    """
+    q = QueryBuilder()
+    q = q[q["strings"] != strng].apply("NEW 1", q["uint32"] * q["int32"] / q["float32"])
+    q = q.apply("NEW 2", q["float32"] + q["float64"] + q["NEW 1"])
+    q = q.apply("NEW 3", q["NEW 2"] - q["NEW 1"] + q["timestamp"])
+    return q
+
+
+def query_resample_minutes() -> QueryBuilder:
+    """
+    Resample query for QueryBuilder memory tests
+    """
+    q = QueryBuilder()
+    return q.resample("min").agg(
+        {"int8" : "min",
+        "int16" : "max",
+        "int32" : "first",
+        "int64" : "last",
+        "uint64" : "sum",
+        "float32" : "mean",
+        "float64" : "sum",
+        "strings" : "count",
+        "bool" : "sum"}
+        )
+
+
+def query_row_range_57percent(size: int) -> QueryBuilder:
+    """
+    Row range query for QueryBuilder memory tests
+    Pass size of dataframe and it will generate random row range
+    """
+    percentage_rows_returned = 0.57
+    start_percentage = random.uniform(0.01, 1.0 - percentage_rows_returned)
```

Comment:
This will probably disable starting from 0-th row. Is this the desired behavior?

The same question applies to `query_date_range_57percent`